CodeGeneration/Sources/SyntaxSupport/AttributeNodes.swift

```diff
@@ -141,6 +141,10 @@ public let ATTRIBUTE_NODES: [Node] = [
             name: "documentationArguments",
             kind: .node(kind: .documentationAttributeArgumentList)
           ),
+          Child(
+            name: "isolatedArguments",
+            kind: .node(kind: .isolatedAttributeArguments)
+          ),
```

Comment:
Instead of adding a new child choice you should be able to reuse the `token` child choice.

Reply:
If you don't mind, I would rather leave room for more structure here.

Reply:
What would that structure look like. We generally prefer to parse attributes like a normal function call now instead of adding dedicated syntax for every possible built-in attribute. So, unless you have concrete ideas for a syntax that can’t be represented by a function call, I’d prefer to stay in that realm.

Reply:
I'm not imagining anything grammatically that wouldn't look somewhat like a function call, but a single token is not exactly the function call grammar.  I just don't want to have to refactor all the code consuming this if we add new forms of the attribute that want more arguments.  We do expect this to grow over time: we're already talking about two different forms (the second being `@isolated(caller)`), so I'm pretty comfortable adding an experimental node in the short term.

I believe this first form (`@isolated(any)`) already technically cannot be parsed as a function call, since `any` is a keyword.  Any variation is likely to be "call-inspired", but not strictly limited to the call grammar.

Reply:
I would prefer to use the call grammar and change `parsePrimaryExpression` to allow keywords when the expression parsing flavor is `attributeArguments`. Something like the following (this currently essentially copy-pastes the identifier parsing code for keywords.

```swift
case nil:
  if flavor == .attributeArguments {
    if pattern.admitsBinding && self.lookahead().isInBindingPatternPosition() {
      let identifier = self.consumeAnyToken(remapping: .identifier)
      let pattern = RawPatternSyntax(
        RawIdentifierPatternSyntax(
          identifier: identifier,
          arena: self.arena
        )
      )
      return RawExprSyntax(RawPatternExprSyntax(pattern: pattern, arena: self.arena))
    }

    return RawExprSyntax(self.parseIdentifierExpression(flavor: flavor))
  }
  return RawExprSyntax(RawMissingExprSyntax(arena: self.arena))
}
```

@rintaro Do you have thoughts. I would like to hear your opinion on this.

Reply:
We're certainly leaning toward to reducing special `AttributeSyntax.Arguments` choices, representing them with generic `LabeledExprListSyntax` syntax, and - as Alex mentioned - making an expression parsing "flavor" for attribute arguments.

>  I'm going to work with it this way in the short term, though.

I think I'm fine with that, but could you make it `@_spi(ExperimentalLanguageFeatures)`? Maybe the current `CodeGeneration` doesn't recognize `Child.experimentalFeature` for `.nodeChoices([Child])` nodes, but implementing it shouldn't be that hard.

Reply:
Probably, prepending `\(choice.apiAttributes)` to here
https://github.com/apple/swift-syntax/blob/e77b169ba11273be5c2f4ea6a22150e5d082ffb6/CodeGeneration/Sources/generate-swift-syntax/templates/swiftsyntax/SyntaxNodesFile.swift#L220

Reply:
I've changed this to use `LabeledExprListSyntax`.